src/main/java/seedu/cafectrl/parser/Parser.java

```diff
@@ -278,26 +281,23 @@ private static Command prepareDelete(Menu menu, String arguments, Ui ui) {
         return new DeleteDishCommand(dishIndex, menu, ui);
     }
 
-    private static Command prepareViewTotalStock(Ui ui) {
-        return new ViewTotalStockCommand(new Pantry(ui), ui);
+    private static Command prepareViewTotalStock(Ui ui, Pantry pantry) {
+        return new ViewTotalStockCommand(pantry, ui);
     }
 
     private static Command prepareBuyIngredient(String arguments, Ui ui, Pantry pantry) {
-        Pattern buyIngredientArgumentsPattern = Pattern.compile(INGREDIENT_ARGUMENT_STRING);
+        Pattern buyIngredientArgumentsPattern = Pattern.compile(BUY_INGREDIENT_ARGUMENT_STRING);
```

Comment:
perhaps you could just use the `INGREDIENT_ARGUEMENT_STRING` regex instead of create a new regex since they both are the same regex

Reply:
Oh oops I am sorry, I understand your intentions now